miasm2/ir/ir.py

```diff
@@ -173,21 +173,22 @@ def merge_multi_affect(self, affect_list):
                 continue
 
             # Find collision
-            e_colision = reduce(
-                lambda x, y: x + y, [e.get_modified_slice() for e in expr_list])
-
+            e_colision = reduce(lambda x, y: x.union(y),
+                                (e.get_modified_slice() for e in expr_list),
+                                set())
             # Sort interval collision
-            known_intervals = sorted([(x[1], x[2]) for x in set(e_colision)])
+            known_intervals = sorted([(x[1], x[2]) for x in e_colision])
 
             # Fill with missing data
-            missing_i = get_missing_interval(known_intervals, 0, e.src.size)
+            missing_i = get_missing_interval(known_intervals, 0, dst.size)
 
-            rest = [(m2_expr.ExprSlice(dst, r[0], r[1]), r[0], r[1])
-                    for r in missing_i]
+            remaining = remaining = ((m2_expr.ExprSlice(dst, *interval),
```

Comment:
I know you're going to hate me but... you have one `remaining = ` more than needed here. Sorry mate, I owe you a good beer for that PR.